src/hotspot/share/nmt/vmatree.hpp

```diff
@@ -167,14 +171,20 @@ class VMATree {
     }
   };
 
-  SummaryDiff register_mapping(position A, position B, StateType state, const RegionData& metadata);
+ private:
+  SummaryDiff register_mapping(position A, position B, StateType state, const RegionData& metadata, bool copy_flag = false);
 
+ public:
   SummaryDiff reserve_mapping(position from, position sz, const RegionData& metadata) {
-    return register_mapping(from, from + sz, StateType::Reserved, metadata);
+    return register_mapping(from, from + sz, StateType::Reserved, metadata, false);
+  }
+
+  SummaryDiff commit_mapping(position from, position sz, const RegionData& metadata, bool copy_flag = false) {
+    return register_mapping(from, from + sz, StateType::Committed, metadata, copy_flag);
   }
 
-  SummaryDiff commit_mapping(position from, position sz, const RegionData& metadata) {
-    return register_mapping(from, from + sz, StateType::Committed, metadata);
+  SummaryDiff uncommit_mapping(position from, position sz, const RegionData& metadata) {
+    return register_mapping(from, from + sz, StateType::Committed, metadata, true);
```

Comment:
As Gerard said: This should be `StateType::Reserved`, clearly this isn't caught by our tests so we do need a test that catches this.

Reply:
New check is added to `NMTVMATreeTest.UseFlagInplace` gTest.